### PR TITLE
user-data: tweak Ubuntu 16.04 user-data

### PR DESCRIPTION
**Pull request by smithfarm**
With this tweak, Ubuntu 16.04 VMs come up with

```
ubuntu@target137074030173:~$ cat /etc/resolv.conf
domain teuthology
search teuthology
nameserver 137.74.29.118
nameserver 213.186.33.99
```

Do you know how to prevent the second nameserver line from being added?

### CODE REVIEW

**Comment by smithfarm** [comment]
Probably the second nameserver is being added when DHCP client communicates with the OVH DHCP server. Do we need this DHCP business at all?

**Comment by smithfarm** [comment]
I tried Ubuntu 14.04 and the `/etc/resolv.conf` there is the same.

**Comment by smithfarm** [comment]
Ubuntu 16.04 test running at http://137.74.29.118:8081/

**Comment by smithfarm** [comment]
@dachary Test passed.

**Comment by dachary** [comment]
Good catch ! What about removing the hardcoded eth0 where possible like so ?

```
diff --git a/teuthology/openstack/openstack-centos-6.5-user-data.txt b/teuthology/openstack/openstack-centos-6.5-user-data.txt
index 6f705a31..27e705df 100644
--- a/teuthology/openstack/openstack-centos-6.5-user-data.txt
+++ b/teuthology/openstack/openstack-centos-6.5-user-data.txt
@@ -2,7 +2,7 @@
 bootcmd:
  - echo nameserver {nameserver} | tee /etc/resolv.conf
  - echo search {lab_domain} | tee -a /etc/resolv.conf
- - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-eth0
+ - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-*
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
  - yum install -y yum-utils && yum-config-manager --add-repo https://dl.fedoraproject.org/pub/epel/6/x86_64/ && yum install --nogpgcheck -y epel-release && rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-6 && rm /etc/yum.repos.d/dl.fedoraproject.org*
diff --git a/teuthology/openstack/openstack-centos-7.0-user-data.txt b/teuthology/openstack/openstack-centos-7.0-user-data.txt
index 2d7b2bce..b7ab7ab5 100644
--- a/teuthology/openstack/openstack-centos-7.0-user-data.txt
+++ b/teuthology/openstack/openstack-centos-7.0-user-data.txt
@@ -2,7 +2,7 @@
 bootcmd:
  - echo nameserver {nameserver} | tee /etc/resolv.conf
  - echo search {lab_domain} | tee -a /etc/resolv.conf
- - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-eth0
+ - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-*
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
  - ( echo ; echo "MaxSessions 1000" ) >> /etc/ssh/sshd_config
diff --git a/teuthology/openstack/openstack-centos-7.1-user-data.txt b/teuthology/openstack/openstack-centos-7.1-user-data.txt
index 2d7b2bce..b7ab7ab5 100644
--- a/teuthology/openstack/openstack-centos-7.1-user-data.txt
+++ b/teuthology/openstack/openstack-centos-7.1-user-data.txt
@@ -2,7 +2,7 @@
 bootcmd:
  - echo nameserver {nameserver} | tee /etc/resolv.conf
  - echo search {lab_domain} | tee -a /etc/resolv.conf
- - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-eth0
+ - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-*
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
  - ( echo ; echo "MaxSessions 1000" ) >> /etc/ssh/sshd_config
diff --git a/teuthology/openstack/openstack-centos-7.2-user-data.txt b/teuthology/openstack/openstack-centos-7.2-user-data.txt
index 2d7b2bce..b7ab7ab5 100644
--- a/teuthology/openstack/openstack-centos-7.2-user-data.txt
+++ b/teuthology/openstack/openstack-centos-7.2-user-data.txt
@@ -2,7 +2,7 @@
 bootcmd:
  - echo nameserver {nameserver} | tee /etc/resolv.conf
  - echo search {lab_domain} | tee -a /etc/resolv.conf
- - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-eth0
+ - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-*
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
  - ( echo ; echo "MaxSessions 1000" ) >> /etc/ssh/sshd_config
diff --git a/teuthology/openstack/openstack-debian-8.0-user-data.txt b/teuthology/openstack/openstack-debian-8.0-user-data.txt
index 8a361700..0d7ae3b2 100644
--- a/teuthology/openstack/openstack-debian-8.0-user-data.txt
+++ b/teuthology/openstack/openstack-debian-8.0-user-data.txt
@@ -3,7 +3,7 @@ bootcmd:
  - apt-get remove --purge -y resolvconf || true
  - echo 'prepend domain-name-servers {nameserver};' | sudo tee -a /etc/dhcp/dhclient.conf
  - echo 'supersede domain-name "{lab_domain}";' | sudo tee -a /etc/dhcp/dhclient.conf
- - ifdown eth0 ; ifup eth0
+ - ifdown -a ; ifup -a
  - grep --quiet {nameserver} /etc/resolv.conf || ( echo 'nameserver {nameserver}' ; echo 'search {lab_domain}' ) | sudo tee /etc/resolv.conf
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
diff --git a/teuthology/openstack/openstack-opensuse-42.1-user-data.txt b/teuthology/openstack/openstack-opensuse-42.1-user-data.txt
index 114f09e0..b815f725 100644
--- a/teuthology/openstack/openstack-opensuse-42.1-user-data.txt
+++ b/teuthology/openstack/openstack-opensuse-42.1-user-data.txt
@@ -2,7 +2,7 @@
 bootcmd:
  - echo nameserver {nameserver} | tee /etc/resolv.conf
  - echo search {lab_domain} | tee -a /etc/resolv.conf
- - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network/ifcfg-eth0
+ - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network/ifcfg-*
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
  - ( echo ; echo "MaxSessions 1000" ) >> /etc/ssh/sshd_config
diff --git a/teuthology/openstack/openstack-sle-12.2-user-data.txt b/teuthology/openstack/openstack-sle-12.2-user-data.txt
index 3cec49e8..a9a97a65 100644
--- a/teuthology/openstack/openstack-sle-12.2-user-data.txt
+++ b/teuthology/openstack/openstack-sle-12.2-user-data.txt
@@ -2,7 +2,7 @@
 bootcmd:
  - echo nameserver {nameserver} | tee /etc/resolv.conf
  - echo search {lab_domain} | tee -a /etc/resolv.conf
- - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network/ifcfg-eth0
+ - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network/ifcfg-*
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
  - ( echo ; echo "MaxSessions 1000" ) >> /etc/ssh/sshd_config
diff --git a/teuthology/openstack/openstack-ubuntu-12.04-user-data.txt b/teuthology/openstack/openstack-ubuntu-12.04-user-data.txt
index 904c4085..6630b914 100644
--- a/teuthology/openstack/openstack-ubuntu-12.04-user-data.txt
+++ b/teuthology/openstack/openstack-ubuntu-12.04-user-data.txt
@@ -3,7 +3,7 @@ bootcmd:
  - apt-get remove --purge -y resolvconf || true
  - echo 'prepend domain-name-servers {nameserver};' | sudo tee -a /etc/dhcp/dhclient.conf
  - echo 'supersede domain-name "{lab_domain}";' | sudo tee -a /etc/dhcp/dhclient.conf
- - ifdown eth0 ; ifup eth0
+ - ifdown -a ; ifup -a
  - grep --quiet {nameserver} /etc/resolv.conf || ( echo 'nameserver {nameserver}' ; echo 'search {lab_domain}' ) | sudo tee /etc/resolv.conf
  - ( curl --silent http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(curl --silent http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
diff --git a/teuthology/openstack/openstack-ubuntu-14.04-user-data.txt b/teuthology/openstack/openstack-ubuntu-14.04-user-data.txt
index d25f6141..450b5256 100644
--- a/teuthology/openstack/openstack-ubuntu-14.04-user-data.txt
+++ b/teuthology/openstack/openstack-ubuntu-14.04-user-data.txt
@@ -3,7 +3,7 @@ bootcmd:
  - apt-get remove --purge -y resolvconf || true
  - echo 'prepend domain-name-servers {nameserver};' | sudo tee -a /etc/dhcp/dhclient.conf
  - echo 'supersede domain-name "{lab_domain}";' | sudo tee -a /etc/dhcp/dhclient.conf
- - ifdown eth0 ; ifup eth0
+ - ifdown -a ; ifup -a
  - grep --quiet {nameserver} /etc/resolv.conf || ( echo 'nameserver {nameserver}' ; echo 'search {lab_domain}' ) | sudo tee /etc/resolv.conf
  - ( wget -qO - http://169.254.169.254/2009-04-04/meta-data/hostname | sed -e 's/[\.-].*//' ; eval printf "%03d%03d%03d%03d.{lab_domain}" $(wget -qO - http://169.254.169.254/2009-04-04/meta-data/local-ipv4 | tr . ' ' ) ) | tee /etc/hostname
  - hostname $(cat /etc/hostname)
```

**Comment by smithfarm** [comment]
@dachary Added two more commits, please have a look.

File: teuthology/openstack/openstack-centos-6.5-user-data.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 bootcmd:
  - echo nameserver {nameserver} | tee /etc/resolv.conf
  - echo search {lab_domain} | tee -a /etc/resolv.conf
- - sed -ie 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-*
+ - sed -i -e 's/PEERDNS="yes"/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-*
```
Comment (dachary) [inline teuthology/openstack/openstack-centos-6.5-user-data.txt]: nice catch !

**Comment by smithfarm** [comment]
"single rados job" test now running (for all five supported distros) at http://149.202.184.202:8081/

**Comment by smithfarm** [comment]
Also ran a "teuthology-lock" check and visually examined `/var/log/cloud-init-output.log` and `/etc/resolv.conf` with the following results:

* Ubuntu 14.04 - OK
* Ubuntu 16.04 - OK
* CentOS 7.2 - OK
* openSUSE 42.1 - OK
* SLE-12-SP2 - OK

ntp setup in user-data is done for openSUSE and SLE, but ceph-cm-ansible should do that for the other distros.

The "single rados job" test passed on four of the five distros. It fails on SLE-12-SP2 due to:

```
2017-02-12T13:28:16.734 INFO:teuthology.orchestra.run.target137074026202:Running: 'sudo chcon system_u:object_r:var_log_t:s0 /home/ubuntu/cephtest/archive/syslog/kern.log'
2017-02-12T13:28:16.871 INFO:teuthology.orchestra.run.target137074026202.stderr:chcon: invalid context: 'system_u:object_r:var_log_t:s0': No such file or directory
```

**Pull request merged**